front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -304,38 +310,56 @@ export default function AssistantBuilder({
   );
 
   const onAssistantSave = async () => {
-    setDisableUnsavedChangesPrompt(true);
-    setIsSavingOrDeleting(true);
-    const res = await submitAssistantBuilderForm({
-      owner,
-      builderState,
-      agentConfigurationId,
-      slackData: {
-        selectedSlackChannels: selectedSlackChannels || [],
-        slackChannelsLinkedWithAgent,
-      },
-    });
-
-    if (res.isErr()) {
-      setIsSavingOrDeleting(false);
-      sendNotification({
-        title: "Error saving Assistant",
-        description: res.error.message,
-        type: "error",
-      });
+    if (!submitEnabled) {
+      // Redirect to the right screen if there are errors.
+      if (instructionsError) {
+        setScreen("instructions");
+      } else if (actionsErrors.length) {
+        setScreen("actions");
+      } else if (assistantHandleError || descriptionError) {
+        setScreen("naming");
+      } else {
+        // Should not happen as it means we had an error without any user facing description.
+        sendNotification({
+          title: "Error saving Assistant",
+          description:
+            "Assistant cannot be saved due to a unknown validation error.",
```

Comment:
There's an error here, it should be "an unknown" instead of "a unknown". Also I'm not sure "validation error" means something for the users. You can check some suggestions here: https://dust.tt/w/0ec9852c2f/assistant/J1eWF712C4

Reply:
👍🏼 nice usage of assistant :)